docs/proposals/interceptors.md

```diff
@@ -0,0 +1,196 @@
+* The `getAlias()` function returns aliases if the current execution filed has aliases. If not, it returns the name of the field.
+
+Users will be able to access these APIs within the interceptor function.
+
+### GraphQL Interceptor
+Interceptors can be defined as a readonly service class that infers the Interceptor object provided by the GraphQL package. User-specific name can be used as the service class name. The Interceptor service class should have the implementation of the `execute()` remote function that infers from the interceptor service object. Apart from that, It can not have any other `resource/remote` methods inside the interceptor. Otherwise, it will be considered the invalid declaration of an interceptor function. However, the users are able to define the usual functions inside the interceptors. Code needed to be included in the interceptor should be kept inside the `execute()` function. To invoke the actual resolver function or the next interceptor the `context.resolve()` function can be called. When calling the `context.resolve()` function, the `graphql:Field` must be passed as the function argument. This will invoke the next interceptor if available. If not, it will call the actual resolver. Following is an example of a GraphQL interceptor.
+
+```ballerina
+readonly service class InterceptorName {
+   *graphql:Interceptor;
+ 
+   remote function execute(graphql:Context context, graphql:Field 'field) returns anydata|error {
+       // Do some work
+       var output = context.resolve('field);
+       // Do some work
+   }
+}
+```
+Interceptors allow alternating the value returning from `context.resolve()` function. When alternating the return value, the new value should be either the same type value as the actual resolver return type or an error. Otherwise, it'll return an error.
+(Ex: If the actual revolver returns a list, the interceptor should return a list. But the list elements can be different.)
+
+### GraphQL Context
+GraphQL [context](https://github.com/ballerina-platform/module-ballerina-graphql/blob/master/docs/spec/spec.md#8-context) is used to pass meta-information among the GraphQL resolver functions. It will be created per each request. The following function will be added to the context in order to control the interceptor execution flow. It accepts the `graphql:Field` object as the argument. The resolver function returns an `anydata` that includes the result of the execution of the next interceptor function.
+
+```ballerina
+public isolated function resolve(graphql:Field ‘field) returns anydata;
+```
+### GraphQL Resolver Configurations
+The resolver config record can be declared in the following format. Since the GraphQL package had not introduced any resolver config earlier, the record includes only the interceptor field. It can accept an array of interceptor instances.
+
+```ballerina
+public type GraphqlResolverConfig record {|
+   readonly readonly & Interceptors[] interceptors = [];
+|};
+```
```

Comment:
We haven't implemented this, right? Why are we including this here?

Reply:
removed!